codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/StalledStreamProtectionConfigCustomization.kt

```diff
@@ -0,0 +1,190 @@
+    )
+
+    override fun section(section: ServiceConfig): Writable {
+        return when (section) {
+            ServiceConfig.ConfigImpl -> writable {
+                rustTemplate(
+                    """
+                    /// Return a reference to the stalled stream protection configuration contained in this config, if any.
+                    pub fn stalled_stream_protection_config(&self) -> #{Option}<&#{StalledStreamProtectionConfig}> {
+                        self.config.load::<#{StalledStreamProtectionConfig}>()
+                    }
+                    """,
+                    *codegenScope,
+                )
+            }
+            ServiceConfig.BuilderImpl -> writable {
+                rustTemplate(
+                    """
+                    /// Set the [`StalledStreamProtectionConfig`](#{StalledStreamProtectionConfig})
+                    /// to configure protection for stalled streams.
+                    pub fn stalled_stream_protection_config(
+                        mut self,
+                        stalled_stream_protection_config: impl #{Into}<#{StalledStreamProtectionConfig}>
+                    ) -> Self {
+                        self.set_stalled_stream_protection_config(#{Some}(stalled_stream_protection_config.into()));
+                        self
+                    }
+                    """,
+                    *codegenScope,
+                )
+
+                rustTemplate(
+                    """
+                    /// Set the [`StalledStreamProtectionConfig`](#{StalledStreamProtectionConfig})
+                    /// to configure protection for stalled streams.
+                    pub fn set_stalled_stream_protection_config(
+                        &mut self,
+                        stalled_stream_protection_config: #{Option}<#{StalledStreamProtectionConfig}>
+                    ) -> &mut Self {
+                        self.config.store_or_unset(stalled_stream_protection_config);
+                        self
+                    }
+                    """,
+                    *codegenScope,
+                )
+            }
+
+            else -> emptySection
+        }
+    }
+}
+
+class StalledStreamProtectionOperationCustomization(
+    private val codegenContext: ClientCodegenContext,
+) : OperationCustomization() {
+    private val rc = codegenContext.runtimeConfig
+
+    override fun section(section: OperationSection): Writable = writable {
+        when (section) {
+            is OperationSection.AdditionalInterceptors -> {
+                val model = codegenContext.model
+                val stalledStreamProtectionModule = RuntimeType.smithyRuntime(rc).resolve("client::stalled_stream_protection")
+                // Only bother mounting this interceptor when an operation's input and/or output
+                // has a blob shape member.
```

Comment:
I think it should always be applied. SImpler non-stream operations can stall too so long as they parse a response body and not just headers.